apps/web/src/components/ReportModal/ReportModal.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { useTranslation } from 'next-i18next';
+
+import { Modal, Button } from 'ui';
+
+type ReportModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  reportProjectClick: (error: string, message: string) => void;
+};
+
+function ReportModal({
+  isOpen,
+  onClose,
+  reportProjectClick,
+}: ReportModalProps) {
+  const [selected, setSelected] = useState(null);
+  const [message, setMessage] = useState('');
+  const [error, setError] = useState(false);
+
+  const { t } = useTranslation('report-modal');
+
+  useEffect(() => {
+    setSelected(false);
+    setMessage('');
+  }, [isOpen]);
+
+  const TAGS = [
+    {
+      name: 'spam',
+      value: 'Spam',
+    },
+    {
+      name: 'nudity',
+      value: 'Nudity',
+    },
+    {
+      name: 'scam',
+      value: 'Scam',
+    },
+    {
+      name: 'illegal',
+      value: 'Illegal',
+    },
+    {
+      name: 'violence',
+      value: 'Violence',
+    },
+    {
+      name: 'suicide-or-self-injury',
+      value: 'Suicide or self injury',
+    },
+    {
+      name: 'hate-speech',
+      value: 'Hate speech',
+    },
+    {
+      name: 'something-else',
+      value: 'Something else',
+    },
+  ];
+
+  return (
+    <Modal
+      open={isOpen}
+      onClose={onClose}
+      modalClassName='bg-grey-dark relative rounded-md w-full max-w-[500px] md:w-auto'
+      contentClassName='p-4 max-h-full md:max-h-screen-md overflow-y-auto'
+    >
+      <div className='flex flex-col gap-2 md:gap-4'>
+        <span className='text-2xl md:text-3xl font-semibold'>
+          {t('report-modal:report')}
+        </span>
+        <p className='font-medium text-lg md:text-xl'>
+          {t('report-modal:reporting-this-post')}
+        </p>
+        <p className='text-sm md:text-base mb-2 md:mb-4'>
+          {t('report-modal:description')}
+        </p>
+
+        <div className='flex flex-row flex-wrap gap-2 md:gap-4 md:mb-6'>
+          {TAGS?.map((tag, idx) => (
+            <span
+              onClick={() => setSelected(tag)}
+              className={classNames(
+                'rounded-full py-1 px-3 cursor-pointer select-none',
+                {
+                  'bg-red-200 text-red-600': selected == tag,
+                  'bg-white text-gray-600': selected !== tag,
+                }
+              )}
+              key={idx}
+            >
+              {t(tag.name)}
+            </span>
+          ))}
+        </div>
+
+        <div className='flex flex-col gap-2 md:gap-4'>
+          <span className='text-lg md:text-xl'>{t('report-modal:reason')}</span>
+          <p>{t('report-modal:help')}</p>
+          <textarea
```

Comment:
can you add this text area to the `ui` library?
you can copy the styles from <FormTextArea/>

Reply:
here